File: spring-web-java/src/main/java/spring/web/java/infrastructure/util/SecurityContextUtil.java
package spring.web.java.infrastructure.util;

import org.springframework.security.core.context.SecurityContextHolder;

public abstract class SecurityContextUtil {

	public static Long getMemberId() {
		Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
		return Long.valueOf(String.valueOf(principal));
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/global/common/SecurityContextUtil.kt
package spring.web.kotlin.global.common

import org.springframework.http.HttpStatus
import org.springframework.security.core.context.SecurityContextHolder
import spring.web.kotlin.global.exception.BaseException

object SecurityContextUtil {
    fun getMemberId() = SecurityContextHolder.getContext().authentication?.principal.toString().toLongOrNull()
        ?: throw BaseException(ResponseMessage.AUTHENTICATION_FAILED, HttpStatus.UNAUTHORIZED)
}